src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int currentCapacity;
+    private int threshold;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        Node<K, V> currentNode = table[getBucket(key)];
+        Node<K, V> newNode = new Node<>(key, value, null);
 
+        if (currentNode == null) {
+            size++;
+            resizeIfNeeded();
+            table[getBucket(key)] = newNode;
+            return;
+        }
+        if (Objects.equals(key, currentNode.key)) {
+            newNode.next = currentNode.next;
+            table[getBucket(key)] = newNode;
+            return;
+        }
+        while (currentNode.next != null) {
+            if (Objects.equals(key, currentNode.next.key)) {
+                newNode.next = currentNode.next.next;
+                currentNode.next = newNode;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        size++;
+        if (resizeIfNeeded()) {
+            put(newNode.key, newNode.value);
+        }
```

Comment:
You should not call put after resize operation
```suggestion
        resizeIfNeeded();
```